src/main/kotlin/nl/avisi/structurizr/site/generatr/GenerateSiteCommand.kt

```diff
@@ -73,14 +82,40 @@ class GenerateSiteCommand : Subcommand(
             refreshLocalClone()
         }
 
-        val branchNames = branches.split(",")
+        val branchNames: MutableList<String> = if (allBranches) {
+            clonedRepository.getBranchNames(excludeBranches.split(",")).toMutableList()
+        } else {
+            branches.split(",").toMutableList()
+        }
+
+        println("Branches : $branchNames")
+
         val workspaceFileInRepo = File(clonedRepository.cloneDir, workspaceFile)
+        val errorBranches: MutableList<String> = emptyList<String>().toMutableList()
+
+        branchNames.forEach { branch ->
+            try {
+                println("Generating diagrams for branch $branch")
+                clonedRepository.checkoutBranch(branch)
+
+                val workspace = createStructurizrWorkspace(workspaceFileInRepo)
+                generateDiagrams(workspace, File(siteDir, branch))
+
+            } catch (e: Exception) {
+                println("Error Generating diagrams for branch $branch")
+                errorBranches.add(branch)
+            }
+        }
```

Comment:
I think I'd prefer to have a clear separation between an "information gathering" phase and a "site/diagram generation" phase. This "diagram generation" phase is in fact both, which makes the code harder to understand.

In the "information gathering" phase, you could just go over all branches to generate a list of branch names for which `createStructurizrWorkspace` does not throw an exception, meaning a valid Structurizr workspace exists.

If a branch is encountered with a non-existing or invalid workspace, a warning should be logged to `stderr`.